contracts/libs/SignatureValidatorV2.sol

```diff
@@ -64,7 +64,8 @@ library SignatureValidator {
 			// s := schnorr signature
 			// parity := public key y-coord parity (27 or 28)
 			// last uint8 is for the Ambire sig mode - it's ignored
-			(bytes32 px, bytes32 e, bytes32 s, uint8 parity,) = abi.decode(sig, (bytes32, bytes32, bytes32, uint8, uint8));
+			sig.trimToSize(sig.length - 1);
+			(bytes32 px, bytes32 e, bytes32 s, uint8 parity) = abi.decode(sig, (bytes32, bytes32, bytes32, uint8));
```

Comment:
@Ivshti This is a change in the solidity code I'm requesting. It just doesn't work otherwise.  
We remove the Ambire sig mode with trimToSize and afterwards proceed to decoding

Reply:
sorry, that was a bug - I did think about it but I totally forgot to fix it; agreed with the change